python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/sfn.py

```diff
@@ -0,0 +1,160 @@
+import time
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union, cast
+
+import boto3
+import dagster._check as check
+from botocore.exceptions import ClientError
+from dagster import MetadataValue, PipesClient
+from dagster._annotations import experimental, public
+from dagster._core.definitions.metadata import RawMetadataMapping
+from dagster._core.definitions.resource_annotation import TreatAsResourceParam
+from dagster._core.errors import DagsterExecutionInterruptedError
+from dagster._core.execution.context.asset_execution_context import AssetExecutionContext
+from dagster._core.execution.context.compute import OpExecutionContext
+from dagster._core.pipes.client import (
+    PipesClientCompletedInvocation,
+    PipesContextInjector,
+    PipesMessageReader,
+)
+from dagster._core.pipes.utils import open_pipes_session
+
+from dagster_aws.pipes.message_readers import PipesCloudWatchMessageReader
+
+if TYPE_CHECKING:
+    from mypy_boto3_stepfunctions import SFNClient
+    from mypy_boto3_stepfunctions.type_defs import (
+        DescribeExecutionOutputTypeDef,
+        StartExecutionInputRequestTypeDef,
+    )
+
+
+@experimental
+class PipesSFNClient(PipesClient, TreatAsResourceParam):
+    """A pipes client for invoking AWS Step Functions.
+
+    Args:
+        context_injector (Optional[PipesContextInjector]): A context injector to use to inject
+            context into the Step Function execution.
+        message_reader (Optional[PipesMessageReader]): A message reader to use to read messages
+            from the Step Function execution. Defaults to :py:class:`PipesCloudWatchMessageReader`.
+        client (Optional[boto3.client]): The boto Step Functions client used to start the execution.
+        forward_termination (bool): Whether to cancel the Step Function execution when the Dagster process receives a termination signal.
+    """
+
+    def __init__(
+        self,
+        context_injector: PipesContextInjector,
```

Comment:
Do we need a custom context injector for this service like we have one for the Lambda Pipes client? 

Typically we would use `PipesEnvContextInjector` in case we can set env variables.

Should we use something else here?